src/dns.rs

```diff
@@ -0,0 +1,510 @@
+use crate::runtime::Runtime;
+use crate::topology::config::GlobalOptions;
+use futures::{Async, Future, Poll};
+use hyper::client::connect::dns::{Name, Resolve};
+use std::error;
+use std::fmt;
+use std::io;
+use std::net::AddrParseError;
+use std::net::ToSocketAddrs;
+use std::net::*;
```

Comment:
minor nit: can we avoid glob imports, I find they make the code a bit harder to follow.